x/go/signal/routine.go

```diff
@@ -117,6 +128,21 @@ func CancelOnExitErr() RoutineOption {
 	}
 }
 
+func WithPanicPolicy(p PanicPolicy) RoutineOption {
+	return func(r *routineOptions) {
+		r.panicPolicy = p
+	}
+}
+
+func WithMaxRestart(maxRestart int) RoutineOption {
+	return func(r *routineOptions) {
+		if r.panicPolicy != Restart {
```

Comment:
I don't think this should panic. Just do an early return.

Reply:
Now that I think about it, this option is somewhat confusing. It makes me think that `WithMaxRestart` sets the number of times to restart when the go-routine exits with an error, not necessarily when the routine panics.

I also don’t think that `Restart` should be a panic policy. The final policy after the "last straw" could still be one of the panic policies (RecoverNoErr, RecoverErr). Why can't you just make the routine restart if `maxRestarts` is greater than 0, and not make `Restart` a panic policy?

Reply:
This is a great idea

Reply:
`CancelOnExitErr` and `CancelOnExit` also used to panic, I will also make that early return.

Reply:
If you make `CancelOnExitErr` and `CancelOnExit` and enum then you won't need to worry about that

Reply:
Right